daffodil-core/src/main/scala/org/apache/daffodil/dsom/RestrictionUnion.scala

```diff
@@ -73,11 +74,17 @@ final class Restriction(xmlArg: Node, val simpleTypeDef: SimpleTypeDefBase)
 
   lazy val baseQName = {
     val baseQNameNodeSeq = xml \ "@base"
-    val baseQNameString = baseQNameNodeSeq.text
-    val tryBaseQName = QName.resolveRef(baseQNameString, xml.scope,
+    val tryBaseQName = QName.resolveRef(baseQNameNodeSeq.text, xml.scope,
       tunable.unqualifiedPathStepPolicy)
-    Assert.invariant(tryBaseQName.isSuccess)
-    tryBaseQName.get
+    val result = tryBaseQName.isFailure && baseQNameNodeSeq.text.contains(" ") match {
```

Comment:
The assert.invariant ensured tryBaseQName was successful, however if tryBaseQName.isFailure but doesn't contain the " ", this still calls tryBaseQName rather than erroring out